ethexe/contracts/src/Router.sol

```diff
@@ -379,36 +380,59 @@ contract Router is IRouter, OwnableUpgradeable, ReentrancyGuardTransient {
 
     /* Helper private functions */
 
-    function _createProgram(bytes32 _codeId, bytes32 _salt) private returns (address) {
-        Storage storage router = _router();
+    function _createProgram(bytes32 _codeId, Storage storage router) private returns (address program) {
         require(router.genesisBlock.hash != bytes32(0), "router genesis is zero; call `lookupGenesisHash()` first");
 
         require(
             router.protocolData.codes[_codeId] == Gear.CodeState.Validated,
             "code must be validated before program creation"
         );
 
-        // Check for duplicate isn't necessary, because `Clones.cloneDeterministic`
-        // reverts execution in case of address is already taken.
-        address actorId =
-            Clones.cloneDeterministic(router.implAddresses.mirrorProxy, keccak256(abi.encodePacked(_codeId, _salt)));
+        address _mirrorAbi = router.implAddresses.mirrorAbi;
 
-        router.protocolData.programs[actorId] = _codeId;
-        router.protocolData.programsCount++;
+        assembly {
+            let contractSize := extcodesize(_mirrorAbi)
 
-        emit ProgramCreated(actorId, _codeId);
+            let ctr := mload(0x40)
+            mstore(ctr, shl(0xa8, 0x6080806040523460155761))
+            let offset := 0x0b
+            mstore8(add(ctr, offset), byte(0x1e, contractSize))
+            mstore8(add(ctr, add(offset, 0x01)), byte(0x1f, contractSize))
```

Comment:
use `(contractSize & 0xff_ff) << 240` and `mstore` instead of 2 `mstore8` and `byte` approach